napari/layers/points/points.py

```diff
@@ -1159,29 +1220,26 @@ def _slice_data(self, indices):
         """
         # Get a list of the data for the points in this slice
         not_disp = list(self.dims.not_displayed)
-        disp = list(self.dims.displayed)
-        indices = np.array(indices)
+        indices = np.array(dims_indices)
         if len(self.data) > 0:
             if self.n_dimensional is True and self.ndim > 2:
                 distances = abs(self.data[:, not_disp] - indices[not_disp])
                 sizes = self.size[:, not_disp] / 2
                 matches = np.all(distances <= sizes, axis=1)
-                in_slice_data = self.data[np.ix_(matches, disp)]
                 size_match = sizes[matches]
                 size_match[size_match == 0] = 1
                 scale_per_dim = (size_match - distances[matches]) / size_match
                 scale_per_dim[size_match == 0] = 1
                 scale = np.prod(scale_per_dim, axis=1)
-                indices = np.where(matches)[0].astype(int)
-                return in_slice_data, indices, scale
+                slice_indices = np.where(matches)[0].astype(int)
+                return slice_indices, scale
```

Comment:
Could we take this opportunity to implement this slicing with scipy.spatial.cKDTree? We've used this in a couple of recent PRs in skimage with 1-2 orders of magnitude speedup relative to linear search. See eg: https://gist.github.com/clementkng/cf0321e2b62daed21189867187e36751

Reply:
@jni I feel like that should be it's own PR

Reply:
very happy to use that instead though!!